src/components/Favorite.jsx

```diff
@@ -13,46 +13,35 @@ function Favorite() {
   const [currentPostLink, setCurrentPostLink] = useState("");
   const [postText, setPostText] = useState("");
   const [posts, setPosts] = useState([]);
-  const [loading, setLoading] = useState(true);
+  const [loading, setLoading] = useState(false);
 
   const rowMode = useContext(RowModeContext);
   const { enqueueSnackbar } = useSnackbar();
 
-  let savedPostsArr = JSON.parse(localStorage.getItem("posts")) || [];
-
   useEffect(() => {
-    const fetchPosts = async () => {
-      const posts = [];
-      const fetchPromises = savedPostsArr.map(async (id) => {
-        try {
-          const docRef = doc(db, "posts", id);
-          const doc = await getDoc(docRef);
-          doc?.data() && posts.push({ id: doc.id, post: doc.data() });
-        } catch (e) {
-          enqueueSnackbar("Error while getting post", {
-            variant: "error",
-          });
-        }
-      });
+    if (auth) {
+      const fetchLikedPosts = async () => {
+        setLoading(true);
+        const collections = collection(db, "posts");
+        console.log(collections);
+        const docSnap = await getDocs(collections);
```

Comment:
isn't getting all the data from `posts` collection and then filtering very costly? This is also why we're using pagination as well?